evalml/tests/conftest.py

```diff
@@ -1291,3 +1291,9 @@ def test_context(
 @pytest.fixture
 def AutoMLTestEnv():
     return _AutoMLTestEnv
+
+
+@pytest.fixture
+def tmpdir_with_cleanup(tmpdir):
```

Comment:
There are other ways to clean up files after a test that don't involve using a new fixture (see [here](https://stackoverflow.com/questions/37563396/deleting-py-test-tmpdir-directory-after-successful-test-case)) but this seems simpler to me

Reply:
why not just use one fixture instead of having one for the tmpdir and one for cleaning up? Are there some tests we don't want the clean up functionality?

Reply:
I thought about this but tbh

```python
@pytest.fixture
def tmpdir(tmpdir):
    yield tmpdir
    tmpdir.remove(ignore_errors=True)
```

looked weird to me. That being said, I will try the following:

```python
@pytest.fixture
def tmpdir(tmp_path):
    dir = py.path.local(tmp_path)
    yield dir
    dir.remove(ignore_errors=True)
```

which I think will work and will involve less code changes!

Reply:
Pushed this up! Thanks for the suggestion - it's working nicely